pkg/controller/core.oam.dev/v1alpha2/application/application_controller.go

```diff
@@ -232,6 +222,7 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 		return r.gcResourceTrackers(logCtx, handler, common.ApplicationWorkflowSuspending, false)
 	case common.WorkflowStateTerminated:
 		logCtx.Info("Workflow return state=Terminated")
+		handler.UpdateApplicationRevisionStatus(logCtx, handler.latestAppRev, false, app.Status.Workflow)
```

Comment:
Wrap this in `doWorkflowFinish`?

Reply:
This is function not always called with doWorkflowFinish.